fix(css): cover auto ratio combo in aspect-ratio (#28805)
mdn/content: files/en-us/web/css/aspect-ratio/index.md

### Values

- `auto`

  - : [Replaced elements](/en-US/docs/Web/CSS/Replaced_element) with an intrinsic aspect ratio use _that_ aspect ratio, otherwise the box has no preferred aspect ratio. Size calculations involving intrinsic aspect ratio always work with the content box dimensions.

- {{cssxref("&lt;ratio&gt;")}}

  - : The box's preferred aspect ratio is the specified ratio of `width` / `height`. If `height` and the preceding slash character are omitted, `height` defaults to `1`. Size calculations involving preferred aspect ratio work with the dimensions of the box specified by `box-sizing`.

> - `auto && ratio`

Reviewer: AFAIK, "Values" lists individual components of the property syntax, not combinations. So I think it would be better to describe what happens when both values are given after the syntac example, but before ### Values, as we do for example in https://developer.mozilla.org/en-US/docs/Web/CSS/margin#syntax.

So we could say, for instance, "This property is specified as one or both of the keyword `auto` or a `<ratio>`. If both are given, then..."

>   - : If the element is a replaced element, like [`<img>`](/en-US/docs/Web/HTML/Element/img), then the preferred `ratio` is used till the content is loaded. After the content is loaded, the `auto` value comes in effect and the natural aspect ratio of the loaded content is used.

Reviewer: ```suggestion
  - : If the element is a [replaced element](/en-US/docs/Web/CSS/Replaced_element), such as [`<img>`](/en-US/docs/Web/HTML/Element/img), then the given `ratio` is used until the content is loaded. After the content is loaded, the `auto` value is applied, so the natural aspect ratio of the loaded content is used.
```

>     If the element is a non-replaced element then the preferred `ratio` is used.

Reviewer: ```suggestion
    If the element is not a replaced element, then the given `ratio` is used.
```

}
```

{{EmbedLiveSample('Exploring aspect-ratio effects with fixed width', '100%', '300px')}}

### Fallback to natural aspect ratio

In this example we are using two`<img>` elements. The first element does not have its `src` attribute set to an image file.

```html
<img src="" /> <img src="plumeria.jpg" />
```

The following code sets `3/2` as the preferred aspect ratio and `auto` as a fallback.

> ```css hidden

Author: ```suggestion
```css
```
Haha! I moved remaining content upwards but forgot to remove the `hidden` tag.

}
```

{{EmbedLiveSample('Exploring aspect-ratio effects with fixed width', '100%', '300px')}}

### Fallback to natural aspect ratio

> In this example we are using two images. The first image doesn't have source.

Reviewer: ```suggestion
In this example we are using two`<img>` elements. The first element does not have its `src` attribute set to an image file.
```

```html
<img src="" /> <img src="plumeria.jpg" />
```

> ```css hidden

Reviewer: I don't think it's a good practice to hide code unless there's a really good reason for it. Really the point of live samples is to show code and result together, so not showing code is just confusing.

img {
  display: inline;
  width: 200px;
  border: 2px dashed red;
  background-color: lime;
  vertical-align: top;
}
```

> Following code sets `3/2` as preferred aspect ratio and `auto` value as a fallback.

Reviewer: ```suggestion
The following code sets `3/2` as the preferred aspect ratio and `auto` as a fallback.
```

```css
img {
  aspect-ratio: 3/2 auto;
}
```

> Note how the first image without replaced content keeps the `3/2` aspect ratio. And the second image after the content is loaded uses the images natural aspect ratio.

Reviewer: ```suggestion
Note how the first image without replaced content keeps the `3/2` aspect ratio, while the second image after the content is loaded uses the image's natural aspect ratio.
```
